hollow/src/main/java/com/netflix/hollow/tools/history/keyindex/HollowOrdinalMapper.java

```diff
@@ -0,0 +1,196 @@
+/*
+ *  Copyright 2016-2019 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+package com.netflix.hollow.tools.history.keyindex;
+
+import com.netflix.hollow.core.index.key.PrimaryKey;
+import com.netflix.hollow.core.memory.encoding.HashCodes;
+import com.netflix.hollow.core.read.HollowReadFieldUtils;
+import com.netflix.hollow.core.read.engine.object.HollowObjectTypeReadState;
+import com.netflix.hollow.core.schema.HollowObjectSchema;
+import com.netflix.hollow.tools.util.ObjectInternPool;
+
+import static com.netflix.hollow.core.HollowConstants.ORDINAL_NONE;
+import java.util.Arrays;
+import java.util.HashMap;
+
+public class HollowOrdinalMapper {
+    private int size = 0;
+    private static final double LOAD_FACTOR = 0.7;
+    
+    private Integer[] ordinalMappings;
```

Comment:
Indexing into an array will be limiting. For e.g. some datasets churn ~5 million modification+removals a day. But maybe ok for the scope of this PR, and something we can scale in future. Or we could consider reusing Hollow's `FixedLengthElementArray` here- it's indexed into using long and it would also be more compact.

Reply:
Can you expand on why indexing into an array will be limiting? I’m not sure
I’m on the same page because the previous implementation also used integer
arrays for ordinal mappings.

Drew specifically suggested I use OA/LP for the ordinal mappings

Reply:
Ah yes I missed that. I was worried about exhausting the Integer.MAX_VALUE (~2B) limit for size of an array in java, which I think translates to how many changes can be indexed for a type.
But actually, as you said that limit was already in place, and in fact since the old impl used a state engine per type key index there was a lower limit of ~500M (2^29 is the max ordinals limit).
So I think add another win to the list, this PR has 4X'ed that limit.